akka-persistence/src/main/scala/akka/persistence/journal/leveldb/LeveldbStore.scala

```diff
@@ -112,6 +112,10 @@ private[persistence] trait LeveldbStore extends Actor with WriteJournalBase with
             batch.delete(keyToBytes(Key(nid, sequenceNr, 0)))
             sequenceNr += 1
           }
+
+          if (this.isInstanceOf[LeveldbCompaction]) {
+            self ! LeveldbCompaction.TryCompactLeveldb(persistenceId, toSeqNr)
+          }
```

Comment:
I had initially mixed the LeveldbCompaction trait into LeveldbStore, however when I run the validatePullRequest task I got a "binary compatibility" error, since SharedLeveldbStore (which mixes in LeveldbStore) is actually a public class. Given the fact that this does not seem to be a really important feature to incorporate into the shared journal (which is only used for testing purposes), I chose to mix LeveldbCompaction into LeveldbJournal (which is package protected) thus conforming with the binary compatibility check. However, given that the compaction functionality is dependent to the sequence number of the last message being deleted, it is necessary for LeveldbStore to trigger the process and pass this info to it, hence the cast was required to do the job. Of course, I am open to suggestions :D